packages/opentelemetry-cloud-trace-propagator/src/CloudPropagator.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  HttpTextPropagator,
+  Context,
+  SetterFunction,
+  GetterFunction,
+  TraceFlags,
+} from '@opentelemetry/api';
+import { decToHex, hexToDec } from 'hex2dec';
+import {
+  setExtractedSpanContext,
+  getParentSpanContext,
+} from '@opentelemetry/core';
+
+/**
+ * This file implements propagation for the Stackdriver Trace v1 Trace Context
+ * format.
+ *
+ * The header specification is:
+ * "X-Cloud-Trace-Context: TRACE_ID/SPAN_ID;o=TRACE_TRUE"
+ * Where:
+ *  {TRACE_ID} is a 32-character hexadecimal value representing a 128-bit
+ *    number. It should be unique between your requests, unless you
+ *    intentionally want to bundle the requests together.
+ *  {SPAN_ID} is the decimal representation of the (unsigned) span ID. It
+ *    should be 0 for the first span in your trace. For subsequent requests,
+ *    set SPAN_ID to the span ID of the parent request.
+ *  {TRACE_TRUE} must be 1 to trace request. Specify 0 to not trace the request.
+ */
+
+/** Header that carries span context across Google infrastructure. */
+export const X_CLOUD_TRACE_HEADER = 'x-cloud-trace-context';
+
+export class CloudPropagator implements HttpTextPropagator {
+  inject(context: Context, carrier: unknown, setter: SetterFunction): void {
+    const spanContext = getParentSpanContext(context);
+    if (!spanContext) return;
+
+    const header = `${spanContext.traceId}/${hexToDec(
+      spanContext.spanId
+    )};o=${spanContext.traceFlags & TraceFlags.SAMPLED}`;
+
+    setter(carrier, X_CLOUD_TRACE_HEADER, header);
+  }
+
+  extract(context: Context, carrier: unknown, getter: GetterFunction): Context {
+    const traceContextHeader = getter(carrier, X_CLOUD_TRACE_HEADER);
+    const traceContextHeaderValue = Array.isArray(traceContextHeader)
+      ? traceContextHeader[0]
+      : traceContextHeader;
+    if (typeof traceContextHeaderValue !== 'string') {
+      return context;
+    }
+    const matches = traceContextHeaderValue.match(
+      /^([0-9a-fA-F]{32})(?:\/([0-9]+))(?:;o=([01]))$/
+    );
+    if (!matches) {
+      return context;
+    }
+    const spanContext = {
```

Comment:
Question: should `isRemote` be set?

See https://github.com/open-telemetry/opentelemetry-js/blob/2d08fcd14df86cb6626ca2966c8b4393b609da77/packages/opentelemetry-api/src/trace/span_context.ts#L41

Note that the B3 Propagator does set it. https://github.com/open-telemetry/opentelemetry-js/blob/e9b2cf9aeb1daf5ffbab800681bfe1cafc636576/packages/opentelemetry-core/test/context/B3Propagator.test.ts#L115

Reply:
`isRemote` is an optional parameter and this field is not being used by anyone in OTEL Js SDK.

Reply:
According to the spec (https://github.com/open-telemetry/opentelemetry-specification/blob/d2665fdde6ae40e8ec08536eeb436c15329c2bad/specification/trace/api.md#span ):

> IsRemote is a boolean flag which returns true if the SpanContext was propagated from a remote parent. When extracting a SpanContext through the Propagators API, its IsRemote flag MUST be set to true, whereas the SpanContext of any child spans MUST have it set to false.

> Please review the W3C specification for details on the Tracestate field.

Reply:
I see. Do you want me to set `isRemote` to `true` all the time?

Reply:
I would like to know your thoughts on this.

Reply:
That's what they do in the B3 Propagator, but maybe @mayurkale22 could shed some light on the topic.

Reply:
Agreed, that is what OpenCensus did. Also, the same thing has been done in [python's implementation ](https://github.com/open-telemetry/opentelemetry-python/pull/819/files#diff-0efddcedadf6e82d94dcb8219d531065R66) and in OTEl js B3 Propagator.

I think we should add the `isRemote` flag when `SpanContext` extracted with a propagator.

Reply:
added isRemote flag